inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/config/model/instrumentation/scope/MethodMatcherSettings.java

```diff
@@ -0,0 +1,50 @@
+package rocks.inspectit.oce.core.config.model.instrumentation.scope;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.NotNull;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Data container for settings which will be used as basis for the {@link rocks.inspectit.oce.core.instrumentation.config.model.InstrumentationScope}
+ * and its method matcher.
+ * <p>
+ * An instance of this class represents a matcher which is matching a set of methods.
+ */
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+public class MethodMatcherSettings extends NameMatcherSettings {
+
+    /**
+     * Enum for access modifiers which can be used in the instrumentation configuration.
+     */
+    public enum AccessModifier {
+        PUBLIC, PROTECTED, PACKAGE, PRIVATE
+    }
+
+    /**
+     * Whether the method is a constructor or not.
+     */
+    @NotNull
+    private Boolean isConstructor = false;
+
+    /**
+     * Whether the method is synchronized or not. If this property is `null` the synchronize keyword will be ignored.
+     */
+    private Boolean isSynchronized;
+
+    /**
+     * The arguments which have to match the method's signature.
+     */
+    private List<String> arguments;
+
+    /**
+     * The methods visibility. On of the specified matcher has to match.
+     */
+    @NotNull
+    private List<AccessModifier> visibility = Arrays.asList(AccessModifier.PUBLIC, AccessModifier.PROTECTED, AccessModifier.PACKAGE, AccessModifier.PRIVATE);

```

Comment:
definitely `PRIVATE` should not be included.. Actually, I am quite up to having only `PUBLIC` as default, decreasing the instrumentation size per default.. @JonasKunz your opinion?

Reply:
Yes, good point. But I would rather go with just removing private from the list instead only providing public. But I'm also fine with just providing public by default.

Reply:
The two alternatives are (A) having either all as default or (B) only have public as default.
Benefit of (A): When the user gives a concrete name in the matcher, it works intuitively. E.g. he gives the name of a private method and he does not need to set anything in Addition to make it work as expected. The downside is if he uses a wildcard matcher he might instrument to many methods.
Benefit of (B): prevents issues when using a wildcard for method matchers whe nthere are very many classes and methods involved.

I think having all visibilities as default is fine ,because when do you usually use wildcard matchers? Mostly in combination with "instrument-only-inherited-methods", where the inherited methods are mostly public or protected anyway.

Reply:
As discussed - keeping ALL as default